lib/logstash/filters/dns/resolv_patch.rb

```diff
@@ -35,12 +35,12 @@ def sender(msg, data, host, port=Port)
 
 # ref: https://github.com/logstash-plugins/logstash-filter-dns/issues/51
 #
-# JRuby versions starting at 9.2.0.0 have a bug caused by an upstream Ruby stdlib resolv.rb bug.
+# JRuby versions starting at 9.2.0.0 and prior to 9.2.9.0 have a bug caused by an upstream Ruby stdlib resolv.rb bug.
 # This bug is essentially the same as the previous above bug where there is a leak between the
 # DNS.allocate_request_id/DNS.free_request_id.
-# This fix is required because starting at JRuby 9.2.0.0 the resolv.rb code was updated from the
+# This fix is required because starting at JRuby 9.2.0.0 and prior to 9.2.9.0 the resolv.rb code was updated from the
 # upstream Ruby stdlib code and the previous patch cannot be applied. Also this fix is better than the previous one.
-if jruby_gem_version >= Gem::Version.new("9.2.0.0")
+if jruby_gem_version >= Gem::Version.new("9.2.0.0") && jruby_gem_version < Gem::Version.new("9.2.9.0")
```

Comment:
We can use the `Gem::Requirement.satisfied_by?`:

```suggestion
if Gem::Requirement.new([">= 9.2.0.0", "< 9.2.9.0"]).satisfied_by?(jruby_gem_version)
```

Reply:
I believe these two are equivalent? is there a case where this could fail? do you think using `satisfied_by?` is preferable?

Reply:
they are equivalent, just less code and that's the purpose of Requirement.satisfied_by?, to avoid reinventing version comparison, that sometimes has edge cases like when using suffixes

Reply:
fair! +1